File: app/src/main/java/at/shockbytes/coins/core/SettingsActivity.java
package at.shockbytes.coins.core;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.transition.Explode;
import android.transition.Slide;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.Window;

import at.shockbytes.coins.fragment.SettingsFragment;


/**
 * @author Martin Macheiner
 *         Date: 27.10.2015.
 */
public class SettingsActivity extends AppCompatActivity {

    public static Intent newIntent(Context context) {
        return new Intent(context, SettingsActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().requestFeature(Window.FEATURE_ACTIVITY_TRANSITIONS);
        getWindow().setExitTransition(new Slide(Gravity.BOTTOM));
        getWindow().setEnterTransition(new Explode());
        if(getSupportActionBar() != null){
            getSupportActionBar().setHomeButtonEnabled(true);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        getFragmentManager().beginTransaction()
                .replace(android.R.id.content, new SettingsFragment())
                .commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        if(item.getItemId() == android.R.id.home){
            supportFinishAfterTransition();
        }

        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/at/shockbytes/coins/ui/activity/SettingsActivity.kt
package at.shockbytes.coins.ui.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import at.shockbytes.coins.dagger.AppComponent
import at.shockbytes.coins.ui.fragment.SettingsFragment
import at.shockbytes.dante.ui.activity.BackNavigableActivity


/**
 * @author Martin Macheiner
 * Date: 27.10.2015.
 */
class SettingsActivity : BackNavigableActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        fragmentManager.beginTransaction()
                .replace(android.R.id.content, SettingsFragment.newInstance())
                .commit()
    }

    override fun injectToGraph(appComponent: AppComponent) {
        // Do nothing here
    }

    companion object {

        fun newIntent(context: Context): Intent {
            return Intent(context, SettingsActivity::class.java)
        }
    }

}
